server/mail/ses.go

```diff
@@ -0,0 +1,95 @@
+package mail
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ses"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+)
+
+type fleetSESSender interface {
+	SendRawEmail(input *ses.SendRawEmailInput) (*ses.SendRawEmailOutput, error)
+}
+
+type sesSender struct {
+	client    fleetSESSender
+	sourceArn string
+}
+
+func getFromSES(e fleet.Email) (string, error) {
+	if e.Config == nil {
+		return "", errors.New("app config is nil")
+	}
+	serverURL, err := url.Parse(e.Config.ServerSettings.ServerURL)
+	if err != nil || len(serverURL.Host) == 0 {
+		return "", fmt.Errorf("failed to parse server url %s err: %w", e.Config.ServerSettings.ServerURL, err)
```

Comment:
really a nit, no need to change it, but when wrapping errors you don't need to prefix the error

```suggestion
		return "", fmt.Errorf("failed to parse server url %s: %w", e.Config.ServerSettings.ServerURL, err)
```